src/libcrun/cgroup.c

```diff
@@ -260,7 +260,7 @@ int
 libcrun_cgroup_enter (struct libcrun_cgroup_args *args, struct libcrun_cgroup_status **out, libcrun_error_t *err)
 {
   __attribute__ ((unused)) pid_t sigcont_cleanup __attribute__ ((cleanup (cleanup_sig_contp))) = -1;
-  cleanup_cgroup_status struct libcrun_cgroup_status *status;
+  cleanup_cgroup_status struct libcrun_cgroup_status *status = NULL;
```

Comment:
This looks obviously correct to me.

However, have you considered using C99 declare-and-initialize style instead of C89?  We've mostly converted ostree to C99 declarations and it *greatly* helps in avoiding this foot-gun.

It'd look like this here e.g.:

```
diff --git a/src/libcrun/cgroup.c b/src/libcrun/cgroup.c
index d696d10..c3fafd3 100644
--- a/src/libcrun/cgroup.c
+++ b/src/libcrun/cgroup.c
@@ -260,7 +260,6 @@ int
 libcrun_cgroup_enter (struct libcrun_cgroup_args *args, struct libcrun_cgroup_status **out, libcrun_error_t *err)
 {
   __attribute__ ((unused)) pid_t sigcont_cleanup __attribute__ ((cleanup (cleanup_sig_contp))) = -1;
-  cleanup_cgroup_status struct libcrun_cgroup_status *status;
   struct libcrun_cgroup_manager *cgroup_manager;
   uid_t root_uid = args->root_uid;
   uid_t root_gid = args->root_gid;
@@ -305,7 +304,7 @@ libcrun_cgroup_enter (struct libcrun_cgroup_args *args, struct libcrun_cgroup_st
     return ret;
 
   /* status will be filled by the cgroup manager.  */
-  status = xmalloc0 (sizeof *status);
+  cleanup_cgroup_status struct libcrun_cgroup_status *status = xmalloc0 (sizeof *status);
   status->manager = args->manager;
 
   ret = cgroup_manager->create_cgroup (args, status, err);
```

Reply:
Picking a random example in ostree, see e.g. https://github.com/ostreedev/ostree/blob/998863e74d18ced5b2faecd1e817b8d2d24ecb5c/src/libostree/ostree-sysroot-deploy.c#L905
which uses declare-and-initialize style.

Reply:
that is a good hint! :-)  I guess it is probably time to move away from c89.  I am using it just for (bad) habit.

I've moved the initialization and pushed a new version.

Reply:
Note that my suggested patch is different in an important way.  With the new patch, we're always calling malloc up front, even if `libcrun_get_cgroup_mode()` fails.  

This doesn't matter *here* I think, but it will in other sections of code where there are dependencies (e.g. variable B depends on variable A, and A requires a function call to compute).

Another way to look at this is it's writing C like Rust where the style is generally always to use `let` inside functions, and not all up front.  (I think modern C++ with `auto` is also like this)